packages/react/src/_popover/Popover.tsx

```diff
@@ -79,43 +83,31 @@ export function usePopover<RT extends ReferenceType = ReferenceType>(
 			// Adds distance between the reference and floating element
 			// https://floating-ui.com/docs/offset
 			offset(offsetOption),
-			// Changes the placement of the floating element in order to keep it in view
-			// https://floating-ui.com/docs/flip
-			flip({ padding: DEFAULT_OFFSET }),
 			// Allows you to change the size of the floating element
 			// https://floating-ui.com/docs/size
+			shift({ padding: MIN_SIDE_GUTTER_WIDTH }),
 			size({
 				padding: DEFAULT_OFFSET, // Prevents the floating element hit the edge of the screen
-				apply({
-					availableWidth,
-					availableHeight: _availableHeight,
-					elements,
-					rects,
-				}) {
-					// Popovers can have a predefined max-height if there is enough room on the screen
-					const maxHeight =
-						maxHeightOption && _availableHeight > maxHeightOption
-							? maxHeightOption
-							: _availableHeight;
-
-					// Minimum acceptable height before `flip` will take over
-					const availableHeight = minHeightOption
-						? Math.max(minHeightOption, maxHeight)
-						: maxHeight;
-
+				apply({ elements, rects }) {
 					Object.assign(elements.floating.style, {
-						maxHeight: `${availableHeight}px`,
+						height: 'max-content',
+						...(!!maxHeight && { maxHeight: `${maxHeight}px` }),
```

Comment:
use `Boolean(maxHeight)` instead of `!!maxHeight`.

The first thing people think when they see `!!maxHeight` is "not max height", the opposite of the actual meaning.

Yes, `!!` is less characters, however `Boolean` is more readable.

Reply:
Actually, I probably took the wrong approach here, now that I think about it. `0` is a valid value, even if it seems it would be a strange path to take. I really need to check whether this is `undefined`. What do you think?

- [ ] Check for `undefined` not falsy

Reply:
yes, that's safer
`maxHeight !== undefined` will work